rs/bitcoin/kyt/src/lib.rs

```diff
@@ -23,9 +23,11 @@ impl From<(Txid, HttpGetTxError)> for CheckTransactionResponse {
     fn from((txid, err): (Txid, HttpGetTxError)) -> CheckTransactionResponse {
         let txid = txid.as_ref().to_vec();
         match err {
```

Comment:
I actually want to comment on L.34-36, but GIthub doesn't let me 👎 
```rust
_ => {
                CheckTransactionIrrecoverableError::InvalidTransaction(format!("{:?}", err)).into()
            }
```
Based on my comment [here](https://github.com/dfinity/ic/pull/1864#discussion_r1791724067) I would thought that also errors of type `HttpGetTxError::TxEncoding` and `HttpGetTxError::TxidMismatch` should also be transient since this means that a provider is returning wrong data, which could be solved by querying another provider. If that's the case, maybe it makes also sense to merge both errors (I'm not sure a caller can meaningfully behave differently between the 2 errors), e.g., ProviderDataError(String)?

Reply:
Good points! Thanks! This should be fixed now. I also added a test for malformed transaction data to ensure it is retriable.